### DISCUSSION TITLE
Help with shallow cloning please

### DISCUSSION THREAD

**Discussion by prateekatam** [go-git/go-git#1146] [Q&A] [answered]
Hi. I'm trying to shallow clone a repository using `Depth: 1` in the clone options of PlainCloneContext. 
However, I observed that, it's not really shallow cloning. When I clone in terminal with `--depth=1`, it enumerates 32 objects whereas in go-git `PlainCloneContext` with Depth 1, it enumerates 256172 objects.

Any help on how to simply shallow clone a repository? I'm looking for depth 1 clone of all branches.

TIA

**Comment by pjbgf** [comment, marked as answer]
@prateekatam by default go-git will download all tags, and so will their objects. In order to avoid that, you need to add `Tags: git.NoTags` to your `CloneOptions`.
